File: Draco/Models/ModelObjects/Division.cs
using System;

namespace ModelObjects
{
/// <summary>
/// Summary description for Division
/// </summary>
	public class Division : IComparable
	{
		public Division()
		{
		}

		public Division(long divisionId, long leagueId, string divisionName, int priority, long accountId)
		{
			AccountId = accountId;
			LeagueId = leagueId;
			Id = divisionId;
			Name = divisionName;
			Priority = priority;
		}

        public long Id
        {
            get;
            set;
        }
	
		public long AccountId
		{
            get;
            set;
        }

		public long LeagueId
		{
            get;
            set;
        }
	
		public string Name
		{
            get;
            set;
        }
	
		public int Priority
		{
            get;
            set;
        }
	
		public int CompareTo(Object o)
		{
			Division d = (Division)o;

			if ( Priority > d.Priority )
				return 1;
			else if ( Priority < d.Priority )
				return -1;
			else 
				return 0;
		}
	}
}

File: draco-nodejs/shared/shared-schemas/division.ts
import { z } from 'zod';
import { extendZodWithOpenApi } from '@asteasolutions/zod-to-openapi';
import { bigintToStringSchema, nameSchema } from './standardSchema.js';
import { SeasonNameSchema } from './seasonBase.js';

extendZodWithOpenApi(z);

export const DivisionNameSchema = z.object({
  id: bigintToStringSchema,
  name: nameSchema,
});

export const DivisionSchema = DivisionNameSchema.extend({
  accountId: bigintToStringSchema,
});

export const DivisionSeasonSchema = z.object({
  id: bigintToStringSchema,
  division: DivisionNameSchema,
  season: SeasonNameSchema.optional(),
  priority: z.number().int(),
});

export const UpsertDivisionSeasonSchema = z.object({
  divisionId: bigintToStringSchema.optional(),
  name: nameSchema.optional(),
  priority: z.coerce.number().int().optional(),
  switchToExistingDivision: z.boolean().optional(),
});

export const UpdateDivisionSeasonResponseSchema = z.object({
  success: z.boolean(),
  conflict: z
    .object({
      existingDivisionId: bigintToStringSchema,
      existingDivisionName: z.string(),
    })
    .optional(),
});

export type DivisionNameType = z.infer<typeof DivisionNameSchema>;
export type DivisionType = z.infer<typeof DivisionSchema>;
export type DivisionSeasonType = z.infer<typeof DivisionSeasonSchema>;
export type UpsertDivisionSeasonType = z.infer<typeof UpsertDivisionSeasonSchema>;
export type UpdateDivisionSeasonResponseType = z.infer<typeof UpdateDivisionSeasonResponseSchema>;
